### ISSUE TITLE
Features proposal

### ISSUE THREAD

**Issue by tomasdev** [Mango/slideout#4]
I understand the idea is to keep it as simply as possible, but it would be nice to have:

1. Ability to have Push Panels instead of Slide/Reveal.
2. Ability to have the navigation appear on top of the content instead of behind.
3. Ability to have it on the right, instead of the left (this might get tricky with overflows).

Would you be accepting PRs for these?

**Comment by danzajdband** [comment]
Tomás, thanks for the proposals. We will accept prs for all of them.

**Comment by phillipadsmith** [comment]
> Ability to have the navigation appear on top of the content instead of behind. This means, animating the hidden menu translateX instead of content, and obviously higher z-index.

@tomasdev Have you started any work on this feature? I also find it a bit distracting to have the content overlap the menu when the transition happens.

**Comment by tomasdev** [comment]
@phillipadsmith nope, haven't had the time. Feel free to do it.

**Comment by phillipadsmith** [comment]
@tomasdev Actually, mucking about with it, I'm not sure it requires more than a z-index: 1 on .slideout-menu   Perhaps, I am missing something?

**Comment by tomasdev** [comment]
@phillipadsmith yes sir, it is z-index + animating the menu instead of animating the content.

**Comment by phillipadsmith** [comment]
> yes sir, it is z-index + animating the menu instead of animating the content.

Is the animation you reference necessary? What I'm seeing here is that simply setting the z-index on .slideout-menu solves the issue I was experiencing with overlapping content. That can be achieved easily without any update to the library.

**Comment by hoscb** [comment]
+1 for Ability to have the navigation appear on top of the content instead of behind. This means, animating the hidden menu translateX instead of content, and obviously higher z-index.
